api/controllers/SignupController.js

```diff
@@ -72,12 +76,33 @@ module.exports = {
 
         // Result from an update is an array
         if (result.length > 0) {
-          return res.json(result[0]);
+          return result[0];
         } else {
           // Result from a create is the user object
-          return res.json(result);
+          return result;
         }
       })
+      .then(function(result) {
+        // Create custom referral url after a sms user has been created
+        // Use a users first name if available or default to SHINE
+        let uniqueString = result.firstName || 'SHINE';
+        uniqueUrl = ReferralCodes.generateCustomUrl(uniqueString);
+        platformSmsId = result.id;
+        return UserUrl.countByCodeLike(`${uniqueUrl}%`);
+      })
+      .then(function(count) {
+        return UserUrl.create({
+          code: uniqueUrl + count,
+          platformSmsId: platformSmsId,
+        });
+      })
+      .then(function(result) {
+        return res.json({
```

Comment:
We'll need to be careful of what we return here, since previously it returned the result of a `User.update`/`User.create` and I'm not sure what other systems may be relying on those return values. Can you check what at least [aurora](https://github.com/shinetext/aurora) might be doing with it?

And if things seem safe, maybe we can return a result of all the objects that got created here?
```
{
  data: {
    user: { /* result of User.create/update */ },
    referral_table_name: { /* result of the UserUrl.create */ }
  }
}
```
... it probably depends on if aurora's actually doing anything with the result or not right now.

And we can leave out the `success` field since that's kind of implicit with the return values and 200 code.